core-libs/setup/ssr/optimized-engine/ssr-optimization-options.ts

```diff
@@ -71,10 +71,28 @@ export interface SsrOptimizationOptions {
    *
    * The value should always be higher than `timeout` and `forcedSsrTimeout`.
    *
-   * Default value is 300 seconds (5 minutes).
+   * Default value is 300000 milliseconds (5 minutes).
    */
   maxRenderTime?: number;
 
+  /**
+   * Instead of immediately falling back to CSR while a render is in progress,
+   * this option will make the subsequent requests wait for the current render,
+   * honoring the `timeout` option.
+   *
+   * E.g., consider the following setup where `timeout` options is set 3s,
+   * and the given request takes 4s to render.
+   * The flow is as follows:
+   *
+   * - 1st request arrives and triggers the SSR.
+   * - 2nd request for the same URL arrives 2s after the 1st one.
+   *    Instead of falling back to CSR, it waits for the render.
+   * - 1st request times out after 3s.
+   * - one second after the timeout, the current render finishes.
+   * - the 2nd request returns SSR after only 2s of waiting.
```

Comment:
```suggestion
   * - 1st request arrives and triggers the SSR.
   * - 2nd request for the same URL arrives 2s after the 1st one.
   *    Instead of falling back to CSR, it waits for the render.
   * - 1st request times out after 3s (and fallbacks to CSR)    // <----- am I right?
   * - one second after the timeout, the current render finishes.
   * - the 2nd request returns SSR after only 2s of waiting.
```